sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlockBlobAsyncApiTests.java

```diff
@@ -0,0 +1,1920 @@
+                                    keyStream.read(strippedKeyBytes);
+                                    SecretKeySpec keySpec = new SecretKeySpec(strippedKeyBytes, CryptographyConstants.AES);
+
+                                    Cipher cipher = Cipher.getInstance("AES/GCM/NoPadding");
+
+                                    int numChunks = (int) Math.ceil(data.array().length / (4 * 1024 * 1024.0));
+
+                                    for (int i = 0; i < numChunks; i++) {
+                                        byte[] iv = new byte[12];
+                                        ciphertextInputStream.read(iv);
+                                        GCMParameterSpec gcmParameterSpec = new GCMParameterSpec(16 * 8, iv);
+
+                                        cipher.init(Cipher.DECRYPT_MODE, keySpec, gcmParameterSpec);
+
+                                        int bufferSize = Math.min(ciphertextInputStream.available(), (4 * 1024 * 1024) + 16);
+                                        byte[] buffer = new byte[bufferSize];
+                                        ciphertextInputStream.read(buffer);
+                                        plaintextOutputStream.write(cipher.doFinal(buffer));
+                                    }
+                                } catch (Exception e) {
+                                    throw new RuntimeException(e);
+                                }
+                                return Mono.zip(Mono.just(plaintextOriginal), Mono.just(plaintextOutputStream.toByteArray()));
+                            });
+                    }));
+
+        StepVerifier.create(response)
+            .assertNext(r -> assertArrayEquals(r.getT1(), r.getT2()))
+            .verifyComplete();
+    }
+
+    private void encryptionTestHelper(int size, int byteBufferCount) {
+        List<ByteBuffer> byteBufferList = new ArrayList<>();
+
+        // Sending a sequence of buffers allows us to test encryption behavior in different cases when the buffers do
+        // or do not align on encryption boundaries.
+        for (int i = 0; i < byteBufferCount; i++) {
+            byteBufferList.add(getRandomData(size));
+        }
+        Flux<ByteBuffer> flux = Flux.fromIterable(byteBufferList);
+
+
+        // Test buffered upload.
+        ParallelTransferOptions parallelTransferOptions = new ParallelTransferOptions()
+            .setBlockSizeLong((long) size)
+            .setMaxConcurrency(2);
+
+        StepVerifier.create(bec.upload(flux, parallelTransferOptions)
+                .then(FluxUtil.collectBytesInByteBufferStream(bec.downloadStream()).map(ByteBuffer::wrap)))
+            .assertNext(outputByteBuffer -> compareListToBuffer(byteBufferList, outputByteBuffer))
+            .verifyComplete();
+    }
+
+    // Requires specific container set up coordinated between languages. Should only be run manually.
+    @Disabled
+    @Test
+    public void testForCrossPlat() {
+        //todo isbr when we have the setup
+    }
```

Comment:
For the synchronous version of this test, does it pass when ran manually? Or does that not have the resource either required to run it? If there are no resources and the sync version do not pass, let's just remove this async test as it is sort of awkward to have a disabled, empty test.

Reply:
the sync version of this test does not work. in the future it may make sense to configure the containers, but for now the async test will be removed.